tests/SpecTests/Operation.php

```diff
@@ -184,6 +184,16 @@ public static function fromCrud(stdClass $operation)
         return $o;
     }
 
+    public static function fromReadWriteConcern(stdClass $operation)
```

Comment:
Unlike `fromCrud`, this method has no mention of `collectionOptions`. Does it actually pass collection options to the collection? Looking at the test json files, the write concern always seems to be the empty document, and I am wondering whether the tests pass because they do not attempt to set the default write concern when performing operations rather than because the driver does not send the empty write concern document to the server.

If the `collectionOptions` do need to be set here, it seems sensible to simply call `fromCrud`.

Reply:
Great catch. I've fixed this and added handling for `databaseOptions` as well, which was previously ignored.